middleware/step_build.go

```diff
@@ -179,3 +179,11 @@ func (h decoratedBuildHandler) HandleBuild(ctx context.Context, in BuildInput) (
 	ctx = RecordMiddleware(ctx, h.With.ID())
 	return h.With.HandleBuild(ctx, in, h.Next)
 }
+
+// BuildHandlerFunc provides a wrapper around a function to be used as a build middleware handler.
+type BuildHandlerFunc func(context.Context, BuildInput) (BuildOutput, Metadata, error)
+
+// HandleBuild invokes the wrapped function with the provided arguments.
+func (b BuildHandlerFunc) HandleBuild(ctx context.Context, in BuildInput) (BuildOutput, Metadata, error) {
+	return b(ctx, in)
+}
```

Comment:
Really minor nit, but to help with future capability might be good to add a small type assert check to ensure that the `XHandlerFunc` satisfies the `XHandler` interface.

Similar to `var _ BuildHandler = (*decoratedBuildHandler)(nil)` above.